### DISCUSSION TITLE
Deprecated PHP grpc stubs (Spiral/GRPC/ServiceInterface is deprecated)

### DISCUSSION THREAD

**Discussion by Bibob7** [spiral/roadrunner-plugins#199] [Q&A]
Hello, my PHPStorm tells me that the used `ServiceInterface` that is used in my generated php stubs is deprecated. I wanted to solve this and build a new version of the stubs, but can not find an up to date version of the protoc-gen-pgp-grpc plugin.

The version is in repository also contains a template with the usage of the deprecated ServiceInterface: https://github.com/spiral/roadrunner-plugins/blob/master/grpc/protoc_plugins/protoc-gen-php-grpc/php/template.go

Is there already an updated version?

**Comment by rustatian** [comment]
Hey @Bibob7. Are you using the RRv2?

**Reply by Bibob7** [reply]
Yes correct

**Reply by rustatian** [reply]
Hmm, ok, I'll redirect this question to our PHP team.
@butschster

**Comment by butschster** [comment]
Hi @Bibob7! Are you using [spiral/php-grpc](https://github.com/spiral/php-grpc) package?

**Reply by Bibob7** [reply]
No, I'm using spiral/roadrunner-grpc

**Reply by butschster** [reply]
> No, I'm using spiral/roadrunner-grpc

You need to recompile your proto file with `protoc-gen-php-grpc` from https://github.com/spiral/roadrunner-binary/releases or just replace `Spiral\GRPC\ServiceInterface` with `Spiral\RoadRunner\GRPC\ServiceInterface` in compiled php files
